packages/store/src/synced.ts

```diff
@@ -35,10 +35,10 @@ export interface SyncedConfig<T> {
   defaultValue: T
 }
 
-export const synced = <T>({key, storage, defaultValue}: SyncedConfig<T>) => {
+export const synced = async <T>({key, storage, defaultValue}: SyncedConfig<T>) => {
   const store = writable<T>(defaultValue)
 
-  sync({key, store, storage})
+  await sync({key, store, storage})
 
   return store
 }
```

Comment:
Let's keep this one synchronous, since it'll be very annoying to use otherwise. We could add a new `Synced<T>` type defined as `Writable<T> & {ready: Promise<void>}` in case callers need to await `sync`